worlds/ahit/__init__.py

```diff
@@ -86,19 +87,28 @@ def generate_early(self):
         if self.is_dw_only():
             return
 
-        # If our starting chapter is 4 and act rando isn't on, force hookshot into inventory
-        # If starting chapter is 3 and painting shuffle is enabled, and act rando isn't, give one free painting unlock
-        start_chapter: ChapterIndex = ChapterIndex(self.options.StartingChapter)
-
-        if start_chapter == ChapterIndex.ALPINE or start_chapter == ChapterIndex.SUBCON:
-            if not self.options.ActRandomizer:
-                if start_chapter == ChapterIndex.ALPINE:
-                    self.multiworld.push_precollected(self.create_item("Hookshot Badge"))
-                    if self.options.UmbrellaLogic:
-                        self.multiworld.push_precollected(self.create_item("Umbrella"))
-
-                if start_chapter == ChapterIndex.SUBCON and self.options.ShuffleSubconPaintings:
+        start_chapter = self.options.StartingChapter
+
+        # Take care of some extremely restrictive starts in other chapters with act shuffle off
+        if start_chapter != ChapterIndex.MAFIA and not self.options.ActRandomizer:
```

Comment:
Confused about the `start_chapter != ChapterIndex.MAFIA` check here because every part inside this condition checks for a specific start chapter anyway

Reply:
This whole block could probably be reformatted like this:

```
if self.options.ActRandomizer:
    # In ActRandomizer, restrictive starts are not a concern because [explanation]
    return

if start_chapter == ChapterIndex.ALPINE:
    ...
elif start_chapter == ChapterIndex.SUBCON:
    ...
elif start_chapter == ChapterIndex.BIRDS:
    ...

Reply:
I didn't do it that way in case I decide to add something below it, but I removed `start_chapter != ChapterIndex.MAFIA`.